### PR TITLE
Modify Workbench GeoMap to use single cell selections and show all rows by default

### LINKED ISSUE #4938: Workbench GeoMap not working Correctly
**Describe the bug**
In edge no matter what rows you have selected it will always show all of the collection objects in GeoMap. If you try using GeoLocate or only selecting some rows it will always show every collection object. 

In 7.9.4 GeoMap only works if you select GeoLocate first and then you can select any number of rows and it will show only the rows you have selected. However if you don't select GeoLocate first even if you only have one row selected it will show all of the collection objects. 

**To Reproduce**

1. Go to workbench 
2. Select a data set with locality information 
3. Select one row 
4. Click on GeoMap 
5. See it shows all collection objects 
6. Click on GeoLocate 
7. Go back to GeoMap 
8. On edge see it still shows all collection objects, on 7.9.4 see it shows only the selected row(s) 

**Expected behavior**
GeoMap should only show the selected rows but you should not have to select GeoLocate first for it to work. In the workbench documentation under GeoMap it also says that "GeoMap plots all the points in your selection on the map" which implies that it should only show the selected rows. 

**Screenshots**
7.9.4: 

https://github.com/specify/specify7/assets/150940840/ea382f8a-0504-4765-b58b-3ff19030c307

Edge: 

https://github.com/specify/specify7/assets/150940840/2530b6df-1862-4d92-97bf-26bb33f04c0c

Please, also fill out the following information manually:

- OS: Windows 11
- Browser: Chrome
- Specify 7 Version: edge
- Database Name: lsumzmammals9823
- Collection name: mammals
- User Name: spadmin

### PR DESCRIPTION
**Pull request by sharadsw**
Fixes #4938

Additional change:
- `getSelectedLocalities()` now has a parameter `defaultSelectAll` to specify the default behavior when no cell is selected

### Checklist

- [x] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [ ] Add automated tests
- [x] Add relevant issue to release milestone

### Testing instructions

- Go to a Workbench dataset with locality information (Eg: MammalData)
- Click on Tools -> GeoMap
- Verify GeoMap works for the following scenarios:
  - [ ] GeoMap shows all rows when no cell is selected
  - [ ] GeoMap shows the correct record when 1 cell is selected
  - [ ] GeoMap shows the correct records when multiple cells are selected
  - [ ] GeoMap shows the correct record when 1 row is selected
  - [ ] GeoMap shows the correct records when multiple rows are selected
- [ ] Additionally, make sure GeoLocate works as expected and has no side effects

### CODE REVIEW

**Review by CarolineDenis — APPROVED** [review]

**Review by emenslin — APPROVED** [review]
### Testing instructions



- Go to a Workbench dataset with locality information (Eg: MammalData)
- Click on Tools -> GeoMap
- Verify GeoMap works for the following scenarios:
  - [x] GeoMap shows all rows when no cell is selected
  - [x] GeoMap shows the correct record when 1 cell is selected
  - [x] GeoMap shows the correct records when multiple cells are selected
  - [x] GeoMap shows the correct record when 1 row is selected
  - [x] GeoMap shows the correct records when multiple rows are selected
- [x] Additionally, make sure GeoLocate works as expected and has no side effects

Looks great!

**Review by Areyes42 — APPROVED** [review]
### Testing instructions

-   Go to a Workbench dataset with locality information (Eg: MammalData)
-   Click on Tools -> GeoMap
-   Verify GeoMap works for the following scenarios:
    -  [x]  GeoMap shows all rows when no cell is selected
    -   [x] GeoMap shows the correct record when 1 cell is selected
    -   [x] GeoMap shows the correct records when multiple cells are selected
    -   [x] GeoMap shows the correct record when 1 row is selected
    -   [x] GeoMap shows the correct records when multiple rows are selected
-  [x]  Additionally, make sure GeoLocate works as expected and has no side effects
***
Everything looks good, great job!

File: specifyweb/frontend/js_src/lib/components/WbToolkit/WbLeafletMap.tsx [hunk 1]
```diff
@@ -24,24 +26,36 @@ export function WbLeafletMap({
...
-    const selection = getSelectedLocalities(
-      hot,
-      dataset.columns,
-      mappings.localityColumns,
-      false
-    );
 
-    if (selection === undefined) return undefined;
+  const [localityPoints, setLocalityPoints] = React.useState<
+    RA<Readonly<Record<string, Field<number | string>>>> | undefined
```
Comment (realVinayak) [inline specifyweb/frontend/js_src/lib/components/WbToolkit/WbLeafletMap.tsx:31]: replace with `RA<IR<Field<number | string>>>`

File: specifyweb/frontend/js_src/lib/components/WbToolkit/WbLeafletMap.tsx [hunk 2]
```diff
@@ -24,24 +26,36 @@ export function WbLeafletMap({
   readonly mappings: WbMapping | undefined;
 }): JSX.Element {
   const [showLeafletMap, openLeafletMap, closeLeafletMap] = useBooleanState();
```
Comment (realVinayak) [inline specifyweb/frontend/js_src/lib/components/WbToolkit/WbLeafletMap.tsx:28]: Remove this now? that is, show the map when localityPoints !== undefined. So, openLeafletMap can be handleOpen (the one you have). closeLeafletMap will be ()=>setLocalityPoints(undefined). Don't see a performance problem (compared to what you have -- you recompute localityPoints anyways)

Comment (sharadsw) [reply specifyweb/frontend/js_src/lib/components/WbToolkit/WbLeafletMap.tsx]: fixed

**Review by melton-jason — APPROVED** [review]
Looks good!

**Pull request merged**